### PR TITLE
Harness fixes

### PR DESCRIPTION
**Pull request by tkkuehn**

### CODE REVIEW

File: afids-auto-apply/workflow/scripts/prob_thres.py [hunk 1]
```diff
@@ -1,116 +1,109 @@
...
-                "image": "/home/greydon/Documents/data/afids-auto/data/OASIS/derivatives/20200301/reg_aladin/sub-0249/anat/sub-0249_space-MNI152NLin2009cAsym_res-1mm_T1w.nii.gz",
-            }
+    # find connected components remaining weighted by probability
+    regions = measure.regionprops(label_image=labels, intensity_image=afid_prob_vol)
+    regions.sort(key=lambda x: x.area, reverse=True)
+
+    area_idxs = []
+    dropped_regions = 0
+    for idx, region in enumerate(regions):
+        if region.area > 100:
```
Comment (kaitj) [inline afids-auto-apply/workflow/scripts/prob_thres.py:48]: Could we include a comment here that indicates what the numbers represent (i.e. 100 voxels?)

File: afids-auto-apply/workflow/scripts/prob_thres.py [hunk 2]
```diff
@@ -1,116 +1,109 @@
...
-                        properties[icomp].label,
-                        properties[icomp].weighted_centroid,
-                    ]
-                )
+        try:
+            afid_info, labels = localize_afid(afid_prob_vol, mask_arr)
+        except ValueError as err:
+            logger.warning("No appropriate region found for AFID %s", afid_num)
+            raise ValueError(
+                f"No appropriate region found for AFID {afid_num} in image " f"{iprob}"
```
Comment (kaitj) [inline afids-auto-apply/workflow/scripts/prob_thres.py]: Could `{iprob}` just go in the first f-string?

File: afids-auto-apply/workflow/scripts/prob_thres.py [hunk 3]
```diff
@@ -144,13 +137,14 @@ def seg_to_fcsv(weighted_centroids, fcsv_template, fcsv_output):
...
-    with open(fcsv_output, "w") as f:
-        f.write("\n".join(line for line in fcsv))
+    with open(fcsv_output, "w", encoding="utf-8") as file_:
+        file_.write("\n".join(line for line in fcsv))
 
 
 if __name__ == "__main__":
     weighted_centroids = seg_prob(
         input_image=snakemake.input.warped_img,
+        input_mask=snakemake.input["mask"],
```
Comment (kaitj) [inline afids-auto-apply/workflow/scripts/prob_thres.py]: Just trying to follow this, why `snakemake.input["mask"]` vs `snakemake.input.mask`?

**Review by kaitj — APPROVED** [review]
A few minor things (see comments), but nothing that should stop this PR from being merged in.

There are also some formatting things and duplicate files for apply vs train that could be discussed and cleaned up if appropriate, but can address that in a separate PR.

File: afids-auto-apply/workflow/scripts/prob_thres.py (hunk 3 above)
Comment (tkkuehn) [reply afids-auto-apply/workflow/scripts/prob_thres.py]: These are equivalent AFAIK, will change to make them consistent.

**Pull request merged**